tests/webapp/api/test_bug_job_map_api.py

```diff
@@ -148,3 +162,17 @@ def test_bug_job_map_delete(webapp, eleven_jobs_stored, test_repository,
         content = json.loads(resp.content)
         assert content == {"message": "Bug job map deleted"}
         assert BugJobMap.objects.count() == 0
+
+
+def test_bug_job_bad_job_id(webapp, test_repository):
+    """
+    test we have graceful error when we pass an invalid job_id
+    """
+    bad_job_id = "aaaa"
+
+    resp = webapp.get(
+            reverse("bug-job-map-list", kwargs={"project": test_repository.name}),
```

Comment:
Thank you for adding a test to cover this other case too - however I think it should be using `bug-job-map-detail` not `bug-job-map-list` (see example in `test_bug_job_map_delete`). With that changed, the test will (correctly) start failing, since `BugJobMapViewSet.retrieve()` doesn't yet handle the `ValueError` gracefully (could you add the check there too?).